packages/modular-scripts/src/buildPackage/index.ts

```diff
@@ -25,239 +24,84 @@ import babel from '@rollup/plugin-babel';
-  const configObject = result.config as TSConfig;
-
-  if (!configObject) {
-    reportTSDiagnostics(':root', [result.error as ts.Diagnostic]);
-    throw new Error('Failed to load Typescript configuration');
-  }
-  Object.assign(typescriptConfig, configObject, {
-    // TODO: should probably include the original exclude in this
-    exclude: distinct([
-      // all TS test files, regardless whether co-located or in test/ etc
-      '**/*.stories.ts',
-      '**/*.stories.tsx',
-      '**/*.spec.ts',
-      '**/*.test.ts',
-      '**/*.e2e.ts',
-      '**/*.spec.tsx',
-      '**/*.test.tsx',
-      '__tests__',
-      '**/dist-cjs',
-      '**/dist-es',
-      '**/dist-types',
-      // TS defaults below
-      'node_modules',
-      'bower_components',
-      'jspm_packages',
-      'tmp',
-      // Casting so that configObject.exclude is set to the correct typing
-      // Since configObject is a index type all values are "any" implicitly.
-      ...((configObject.exclude as string[]) || []),
-    ]),
-  });
-
-  typescriptConfig.compilerOptions = typescriptConfig.compilerOptions || {};
-
-  Object.assign(typescriptConfig.compilerOptions, {
-    declarationDir: outputDirectory,
-    noEmit: false,
-    declaration: true,
-    emitDeclarationOnly: true,
-    incremental: false,
-  });
-}
-
-function getPackageEntryPoints(packagePath: string) {
-  const packageJson = packageJsonsByPackagePath[packagePath];
-
-  let compilingBin = false;
-  let main: string | undefined;
-
-  if (packageJson.main) {
-    main = packageJson.main;
-  } else {
-    if (packageJson.bin) {
-      const bins: string[] = Object.values(packageJson.bin) as string[];
-      if (bins.length === 1) {
-        compilingBin = true;
-        main = bins[0];
-      } else {
-        throw new Error(
-          `package.json at ${packagesRoot}/${packagePath} contains multiple "bin" values, bailing...`,
-        );
-      }
-    } else {
-      throw new Error(
-        `package.json at ${packagesRoot}/${packagePath} does not have a "main" or "bin" field, bailing...`,
-      );
-    }
-  }
-
-  return { main, compilingBin };
-}
-
 async function makeBundle(
   packagePath: string,
   preserveModules: boolean,
 ): Promise<boolean> {
+  const {
+    packageNames,
+    packageJsons,
+    publicPackageJsons,
+    packageJsonsByPackagePath,
+    rootPackageJsonDependencies,
+  } = getPackageMetadata();
+
   const logger = getLogger(packagePath);
 
   const packageJson = packageJsonsByPackagePath[packagePath];
 
   const { main, compilingBin } = getPackageEntryPoints(packagePath);
 
   if (!packageJson) {
-    throw new Error(
-      `no package.json in ${packagesRoot}/${packagePath}, bailing...`,
-    );
+    throw new Error(`no package.json in ${packagePath}, bailing...`);
   }
   if (packageJson.private === true) {
-    throw new Error(
-      `${packagesRoot}/${packagePath} is marked private, bailing...`,
-    );
+    throw new Error(`${packagePath} is marked private, bailing...`);
```

Comment:
This should already be checked in actionPreflightCheck with `isValidModularRootPackageJson` (I think you need to rebase)